mitosheet/mitosheet/tests/test_column_definitions.py

```diff
@@ -0,0 +1,253 @@
+
+
+def test_create_backend_with_column_definitions_errors_on_invalid_color():
+    column_definitions= [
+        [
+            {
+                'columns': ['A'],
+                'conditional_formats': [{
+                    'filters': [{'condition': 'less_than_or_equal', 'value': 0}], 
+                    'font_color': 'ABC', 
+                    'background_color': '#000000' 
+                }] 
+            },
+        ]
+    ]
+
+    with pytest.raises(ValueError) as e_info:
+        mito_backend = get_mito_backend(df, column_definitions=column_definitions)
+        
+    assert "ABC" in str(e_info)
+
+def test_create_backend_with_column_definitions_errors_on_no_colors():
+    column_definitions= [
+        [
+            {
+                'columns': ['A'],
+                'conditional_formats': [{
+                    'filters': [{'condition': 'less_than_or_equal', 'value': 0}],
+                }] 
+            },
+        ]
+    ]
+
+    with pytest.raises(ValueError) as e_info:
+        mito_backend = get_mito_backend(df, column_definitions=column_definitions)
+        
+    assert "column_definititon has invalid conditional_format rules. It must set the font_color, background_color, or both." in str(e_info)
+
+def test_create_backend_with_column_definitions_errors_more_formatted_sheets_then_dataframes():
+    column_definitions= [
+        [
+            {
+                'columns': ['A'],
+                'conditional_formats': [{
+                    'filters': [{'condition': 'less_than_or_equal', 'value': 0}],
+                    'font_color': '#000000', 
+                    'background_color': '#000000' 
+                }] 
+            },
+        ], 
+        [
+            {
+                'columns': ['A'],
+                'conditional_formats': [{
+                    'filters': [{'condition': 'less_than_or_equal', 'value': 0}],
+                    'font_color': '#000000', 
+                    'background_color': '#000000' 
+                }] 
+            },
+        ], 
+    ]
+
+    with pytest.raises(ValueError) as e_info:
+        mito_backend = get_mito_backend(df, column_definitions=column_definitions)
+        
+    assert "dataframes are provided." in str(e_info)
+
+def test_create_backend_with_column_definitions_errors_if_column_doesn_not_exist():
+    column_definitions= [
+        [
+            {
+                'columns': ['D'],
+                'conditional_formats': [{
+                    'filters': [{'condition': 'less_than_or_equal', 'value': 0}],
+                    'font_color': '#000000', 
+                    'background_color': '#000000' 
+                }] 
+            },
+        ], 
+    ]
+
+    with pytest.raises(ValueError) as e_info:
+        mito_backend = get_mito_backend(df, column_definitions=column_definitions)
+        
+    assert "don't exist in the dataframe." in str(e_info)
+
+
+def test_create_backend_with_column_definitions_errors_if_invalid_filter_condition():
```

Comment:
Can we parameterize all these tests to make them easier to maintain? Just a parameter for column, the conditional formats, and then the error you expect out of them.